Gulp csv2json

kamataryo opened the pull request:
#9 #10 #11 
csvからjsonを作るGulpプラグインを追加しました。
こんな感じで使います。

```
const gulp = require('gulp')
const csv2json = require('./lib/wnew_csv2json')

gulp.task('csv2json', () => {
  gulp.src('./example/**/*.csv')
    .pipe(csv2json())
    .pipe(gulp.dest('./dest'))
})
```

複数のcsvファイルソースから、 `menu.json`（メニュー項目名の配列）と `串本.json`、 `公衆トイレ.json`などを作ります。
`npm install`した後に、 `npm run convert`するとサンプルのCSVファイルからJSONファイルを作ります。

Commit:
Prepare gulp plugin development

with test and empty buffer reading function.

Commit:
Prepare gulp plugin development

with test and empty buffer reading function.

Commit:
Merge branch 'gulp-csv2json' of github.com:wakayama-hacker/wnew into gulp-csv2json

Commit:
Enable csv2json in case of single CSV file and single menu

Commit:
Add case with mult menu

Commit:
Add case of several menus

Commit:
Fix bug for nth assert

0 should be first, 2 should be third

Commit:
Fix typo and bug

Commit:
Add test cases for several CSV input, but skipped

Commit:
Fix filename, specified to this project

Commit:
Refactor and comment more

Commit:
Update cmments and coding style

Commit:
Add fix and tests for abnormal input cases

Commit:
Add a assertion library for gulp plugin test

Commit:
Add other libraries

Commit:
Fix bug for cases if menu header is in behind, not in index=0

Commit:
Add more test cases for lacking menu value

Commit:
Add options

Commit:
Make it acceptable with multiple CSV files input

Commit:
Add sample csv data

Commit:
Fix bug close#11

Make it enable to pick up data in several file and has same menu column

Commit:
Add through for the custom gulp plugin

Commit:
Add vinyl

Commit:
Add documents and comments

miya0001 commented on package.json:
```diff
@@ -5,8 +5,10 @@
+    "convert": "gulp csv2json",
     "build": "riot tags/ js/tags.js && gulp build",
-    "test": "eslint --ext .js,.tag . && npm run build && karma start"
+    "test": "eslint --ext .js,.tag . && mocha tests/lib && npm run build && karma start",
```
npm run buildって mochaの前にやった方がいくないですか？

miya0001 replied:
あと`mocha tests/lib`のとこは`npm run test:unit`って書いた方がいいですね。

miya0001 commented on package.json:
```diff
@@ -5,8 +5,10 @@
   "main": "index.html",
   "scripts": {
     "start": "http-server -io",
+    "convert": "gulp csv2json",
```
これ`npm run build`にまとめてください。

miya0001 commented:
`example/` は`data`ていう名前にかえて、そこに放り込んでください。
glupはデフォルトでそこに見にいくようにしていただけると。
細かくてすいませんが `dest` っていうのも通常はコンパイル後のjsとかを放り込むイメージなので、`json`とかに変えてください。（.gitignoreへの追加もお願いします。）

フォークして以下のコマンドで「わー動いた動いた」っていう体験をしてもらえるようにしたいです。
```
npm install
npm run build
npm start
```

miya0001 reviewed (commented):
以下のコメントを参照願います。

https://github.com/wakayama-hacker/wnew/pull/12#issuecomment-282478362

Commit:
Fix build process

destination folders and its source,
clean up npm scripts

kamataryo committed:
Update README.md

Fit to actual folder structure

Commit:
Simplify build process

Make them together in gulp build

Commit:
Fix gulp plugin stream process

CSV should be parsed on each file entered, not at flushing

Commit:
Fix gulp plugin's option key name

column separator, row separator, category(default as 'menu') and extension(default as 'json')

Commit:
Refactor test

update assertion helper, simplify and improve readability of each test cases.

Commit:
Merge branch 'gulp-csv2json' of github.com:wakayama-hacker/wnew into gulp-csv2json

Commit:
Update readme for data maintainers

kamataryo commented:
ワークフローの修正を追加しました。
  + csvファイルは `./data`以下に置くようにし、データのメンテナーにはREADMEから誘導
  + JSONファイルは `./json`以下に出力される
  + データの変換は `npm run build`の内部で実行されるよう修正
  + ユニットテストに先立ってビルドが実行されるよう修正

miya0001 commented:
ありがとうございます！
